sopel_modules/twitter/twitter.py

```diff
@@ -149,29 +164,41 @@ def format_time(bot, trigger, stamp):
         bot.db, bot.config, tz, trigger.nick, trigger.sender, parsed)
 
 
-@module.url(r'https?://(?:m(?:obile)?\.)?twitter\.com/(?P<user>\w+)(?:/status/(?P<status>\d+))?')
-@module.url(r'https?://(?:m(?:obile)?\.)?twitter\.com/i/web/status/(?P<status>\d+).*')
-def get_url(bot, trigger, match):
-    things = match.groupdict()
-    user = things.get('user', None)
-    status = things.get('status', None)
+def _twitter_alt_domains(
+    path_regex: str,
+) -> Callable[[Config], List[re.Pattern]]:
+    """Build a url_lazy loader for the specified callback type.
 
-    if status:
-        output_status(bot, trigger, status)
-    elif user:
-        output_user(bot, trigger, user)
-    else:
-        # don't know how to handle this link; silently fail
-        # explicit is better than implicit
-        return
+    :param path_regex: The path to be appended to each domain regex
+    :return: A loader to be called by url_lazy()
+    """
+    def loader(settings: Config) -> List[re.Pattern]:
+        return [
+            re.compile(r"https?://{}/{}".format(domain, path_regex))
+            for domain in settings.twitter.alternate_domains
+        ]
+
+    return loader
```

Comment:
Function annotations aren't going to work on Python 2; they weren't part of the Python syntax yet. Keeping this as-is would mean the plugin becomes unusable to Sopel 7 users still using Python 2.7, which that version of the bot still supports.

Reply:
:sob:

Reply:
Can `typing` types be used in docstring rtype like this?

Reply:
It's just text. Nothing actually processes the docstrings in this plugin into Sphinx docs (Yet™ at least).